quinn-proto/src/connection/send_buffer.rs

```diff
@@ -415,48 +449,62 @@ mod tests {
         );
     }
 
+    /// tests that large segments are copied as-is in the SendBuffer
     #[test]
-    fn multiple_segments() {
-        let mut buf = SendBuffer::new();
-        const MSG: &[u8] = b"Hello, world!";
-        const MSG_LEN: u64 = MSG.len() as u64;
-
-        const SEG1: &[u8] = b"He";
-        buf.write(SEG1.into());
-        const SEG2: &[u8] = b"llo,";
-        buf.write(SEG2.into());
-        const SEG3: &[u8] = b" w";
-        buf.write(SEG3.into());
-        const SEG4: &[u8] = b"o";
-        buf.write(SEG4.into());
-        const SEG5: &[u8] = b"rld!";
-        buf.write(SEG5.into());
-
-        assert_eq!(aggregate_unacked(&buf), MSG);
-
-        assert_eq!(buf.poll_transmit(16), (0..8, true));
-        assert_eq!(buf.get(0..5), SEG1);
-        assert_eq!(buf.get(2..8), SEG2);
-        assert_eq!(buf.get(6..8), SEG3);
-
-        assert_eq!(buf.poll_transmit(16), (8..MSG_LEN, true));
-        assert_eq!(buf.get(8..MSG_LEN), SEG4);
-        assert_eq!(buf.get(9..MSG_LEN), SEG5);
+    fn multiple_large_segments() {
+        fn dup(data: &[u8]) -> Bytes {
+            let mut buf = BytesMut::with_capacity(data.len() * N);
```

Comment:
nit: it confused me that N is defined later, wasn't sure if I missed something and went scrolling right to the top of the diff looking for it. Why not move the consts to the start of the test?

Reply:
Yeah, did so already.